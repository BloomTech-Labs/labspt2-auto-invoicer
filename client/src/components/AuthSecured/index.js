import React from "react";

// react router dom
import { Link } from "react-router-dom";
import { CompanyConsumer } from "../../contexts/CompanyContext";

const AuthSecured = props => {
  return (
    <CompanyConsumer>
      {({companyState: {credits, unlimited_tier}}) => {
        const plan = unlimited_tier ? "1 Month Unlimited" : credits
        return (
          <ul className="auth-container">
            <p className="credits">Credits: {plan}</p>
            <Link to="/" id="signOut" onClick={props.signOut}>
              Sign Out
            </Link>
          </ul>
        )
      }}
    </CompanyConsumer>
    
  );
};

export default AuthSecured;
