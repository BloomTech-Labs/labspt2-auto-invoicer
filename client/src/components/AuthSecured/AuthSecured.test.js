import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import AuthSecured from "../AuthSecured";

describe("<AuthSecured />", () => {
  it("renders Auth Secured components", () => {
    const { asFragment } = render(
      <Router>
        <AuthSecured />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
