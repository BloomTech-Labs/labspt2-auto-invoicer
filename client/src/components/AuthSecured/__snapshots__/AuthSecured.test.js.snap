// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthSecured /> renders Auth Secured components 1`] = `
<DocumentFragment>
  <ul
    class="auth-container"
  >
    <p
      class="credits"
    >
      Credits: 
    </p>
    <a
      href="/"
      id="signOut"
    >
      Sign Out
    </a>
  </ul>
</DocumentFragment>
`;
